### DISCUSSION TITLE
How to use search on sub queries

### DISCUSSION THREAD

**Discussion by Unisolgr14** [Power-Components/livewire-powergrid#629] [Q&A]
I have many powergrid tables. And, I have to use lots of sub queries using joins as well as not using it. one of them is given below. How to use search/filter through `full_name`
The query extracted from the `datasource()` function

`return ShareInfo::query()
            ->join('share_details','share_details.id','=','share_infos.share_type')
        ->join('parameters','parameters.para_id','=','share_details.share_type')
        ->join('members','members.member_id','=','share_infos.member_id')
        ->where('parameters.para_type','=','share')
        ->select(
        DB::raw("CONCAT(first_name,if(trim(middle_name)='','',concat(' ',middle_name)),if(trim(last_name)='','',concat(' ',last_name)))as     full_name"),
        'share_infos.share_id',
        'share_infos.amount',
        'share_infos.quantity',
        'share_infos.spd',
        'share_infos.share_type',
        'share_infos.status',
        'parameters.para_name as share_type_name') ;`

**Comment by ashek1412** [comment]
I got the same problem

**Comment by marineusde** [comment]
Maybe this helps, cause you can add a QueryBuilder as subquery in an EloquentBuilder:

```
public function datasource(): null|QueryBuilder|EloquentBuilder
{
    return User::query()
            ->from(DB::table(function ($query) {
                $query->selectRaw("users.*, CONCAT(last_name, ', ', first_name) AS full_name")
                    ->from('users');
            }), 'users')
}
```

**Comment by technoholicpl** [comment]
Try this (works for me):

```
public function fields(): PowerGridFields
    {
        return PowerGrid::fields()
            ->add('full_name', function (User $model) {
                return $model->first_name . ' ' . $model->last_name;
            }));
    }
 public function columns(): array
    {
        return [
            Column::add()
                ->title('Full name')
                ->field('full_name','last_name')
                ->searchableRaw("concat(first_name, ' ', last_name)")
                ->sortable(),
   ];
    }
```
